simulator-core/src/test/java/com/evbox/everon/ocpp/simulator/station/component/OCPPCommCtrlComponentTest.java

```diff
@@ -0,0 +1,192 @@
+        assertThat(validationResult.getSetVariableDatum().getAttributeType()).isEqualTo(SetVariableDatum.AttributeType.ACTUAL);
+        assertThat(validationResult.getResult().getAttributeStatus()).isEqualTo(SetVariableResult.AttributeStatus.ACCEPTED);
+    }
+
+    @Test
+    void shouldValidateIfVariableExists() {
+        //given
+        String componentName = OCPPCommCtrlrComponent.NAME;
+        String variableName = "UnknownVariableName";
+        SetVariableDatum.AttributeType attributeType = SetVariableDatum.AttributeType.ACTUAL;
+        int variableValue = StationConstants.DEFAULT_HEARTBEAT_INTERVAL;
+
+        Component component = new Component().withName(new CiString.CiString50(componentName));
+        Variable variable = new Variable().withName(new CiString.CiString50(variableName));
+
+        SetVariableDatum setVariableDatum = new SetVariableDatum()
+                .withComponent(component)
+                .withVariable(variable)
+                .withAttributeType(attributeType)
+                .withAttributeValue(new CiString.CiString1000(String.valueOf(variableValue)));
+
+        //when
+        SetVariableValidationResult validationResult = ocppCommCtrlrComponent.validate(setVariableDatum);
+
+        //then
+        assertCiString(validationResult.getSetVariableDatum().getComponent().getName()).isEqualTo(componentName);
+        assertCiString(validationResult.getSetVariableDatum().getVariable().getName()).isEqualTo(variableName);
+        assertCiString(validationResult.getSetVariableDatum().getAttributeValue()).isEqualTo(String.valueOf(variableValue));
+        assertThat(validationResult.getSetVariableDatum().getAttributeType()).isEqualTo(SetVariableDatum.AttributeType.ACTUAL);
+        assertThat(validationResult.getResult().getAttributeStatus()).isEqualTo(SetVariableResult.AttributeStatus.UNKNOWN_VARIABLE);
+    }
+
+    private void initGetHeartbeatVariableMock(String componentName, String variableName, GetVariableDatum.AttributeType attributeType, String returnValue) {
+        Component component = new Component().withName(new CiString.CiString50(componentName));
+        Variable variable = new Variable().withName(new CiString.CiString50(variableName));
+        CiString.CiString1000 attributeValue = new CiString.CiString1000(returnValue);
+        GetVariableResult.AttributeType returnAttributeType = GetVariableResult.AttributeType.fromValue(attributeType.value());
+
+        given(heartbeatIntervalVariableAccessorMock.get(any(Component.class), any(Variable.class), any(GetVariableDatum.AttributeType.class)))
+                .willReturn(new GetVariableResult()
+                        .withComponent(component)
+                        .withVariable(variable)
+                        .withAttributeType(returnAttributeType)
+                        .withAttributeValue(attributeValue));
+    }
+
+
+}
```

Comment:
lint: Newline at EOF also two consecutive vertical white spaces above ^

Please run this file through a formatter.